### PR TITLE
check quiz disable instead ofblocked = quiz on mining disable check

### PR DESCRIPTION
**Pull request by ice-cronus**
Linked with eskimo [#121](https://github.com/ice-blockchain/eskimo/pull/121)

### CODE REVIEW

File: tokenomics/kyc.go [hunk 1]
```diff
@@ -98,9 +98,10 @@ func (r *repository) validateKYC(ctx context.Context, userID string, state *getC
 		return errors.Wrapf(err, "failed to overrideKYCStateWithEskimoKYCState for %#v", state)
 	}
 	if (state.KYCStepBlocked == users.FacialRecognitionKYCStep && r.isKYCEnabled(ctx, state.LatestDevice, users.FacialRecognitionKYCStep)) ||
-		(state.KYCStepBlocked == users.QuizKYCStep && r.isKYCEnabled(ctx, state.LatestDevice, users.QuizKYCStep)) {
+		((state.KYCStepBlocked == users.QuizKYCStep || state.KYCQuizDisabled) && r.isKYCEnabled(ctx, state.LatestDevice, users.QuizKYCStep)) {
 		return terror.New(ErrMiningDisabled, map[string]any{
 			"kycStepBlocked": state.KYCStepBlocked,
+			"quizDisabled":   state.KYCQuizDisabled,
```
Comment (ice-ares) [inline tokenomics/kyc.go]: No

File: tokenomics/kyc.go [hunk 2]
```diff
@@ -98,9 +98,10 @@ func (r *repository) validateKYC(ctx context.Context, userID string, state *getC
 		return errors.Wrapf(err, "failed to overrideKYCStateWithEskimoKYCState for %#v", state)
 	}
 	if (state.KYCStepBlocked == users.FacialRecognitionKYCStep && r.isKYCEnabled(ctx, state.LatestDevice, users.FacialRecognitionKYCStep)) ||
-		(state.KYCStepBlocked == users.QuizKYCStep && r.isKYCEnabled(ctx, state.LatestDevice, users.QuizKYCStep)) {
+		((state.KYCStepBlocked == users.QuizKYCStep || state.KYCQuizDisabled) && r.isKYCEnabled(ctx, state.LatestDevice, users.QuizKYCStep)) {
 		return terror.New(ErrMiningDisabled, map[string]any{
 			"kycStepBlocked": state.KYCStepBlocked,
```
Comment (ice-ares) [inline tokenomics/kyc.go]: U need to still return kyc step blocked 4 cuz thats what the app expects

**Review by ice-ares — CHANGES_REQUESTED** [review]

File: tokenomics/kyc.go (hunk 2 above)
Comment (ice-cronus) [reply tokenomics/kyc.go]: I told with FE they said they rely on quizStatus.disabled, but ok

Comment (ice-ares) [reply tokenomics/kyc.go]: U didnt ask the proper question

**Pull request merged**